tests/detect_aws_credentials_test.py

```diff
@@ -1,38 +1,103 @@
 import pytest
 
+from pre_commit_hooks.detect_aws_credentials import get_aws_credential_files_from_env
+from pre_commit_hooks.detect_aws_credentials import get_aws_secrets_from_env
+from pre_commit_hooks.detect_aws_credentials import get_aws_secrets_from_file
 from pre_commit_hooks.detect_aws_credentials import main
 from testing.util import get_resource_path
 
 
+def test_get_aws_credentials_file_from_env(monkeypatch):
+    """Test that reading credential files names from environment variables works."""
+    monkeypatch.delenv('AWS_CREDENTIAL_FILE', raising=False)
+    monkeypatch.delenv('AWS_SHARED_CREDENTIALS_FILE', raising=False)
+    monkeypatch.delenv('BOTO_CONFIG', raising=False)
+    assert get_aws_credential_files_from_env() == set()
+    monkeypatch.setenv('AWS_CREDENTIAL_FILE', '/foo')
+    assert get_aws_credential_files_from_env() == {'/foo'}
+    monkeypatch.setenv('AWS_SHARED_CREDENTIALS_FILE', '/bar')
+    assert get_aws_credential_files_from_env() == {'/foo', '/bar'}
+    monkeypatch.setenv('BOTO_CONFIG', '/baz')
+    assert get_aws_credential_files_from_env() == {'/foo', '/bar', '/baz'}
+    monkeypatch.setenv('AWS_CONFIG_FILE', '/xxx')
+    assert get_aws_credential_files_from_env() == {'/foo', '/bar', '/baz', '/xxx'}
+    monkeypatch.setenv('AWS_DUMMY_KEY', 'foobar')
+    assert get_aws_credential_files_from_env() == {'/foo', '/bar', '/baz', '/xxx'}
```

Comment:
to me this feels like 5-6 different tests.  You may find it easier to write this test with `mock.patch.dict` as such (and I would suggest it since it definitely improves readability):


```python
@pytest.mark.parametrize(
    ('environ', 'expected'),
    (
        ({'AWS_CREDENTIAL_FILE': '/foo'}, {'/foo'}),
        # etc. ...
    ),
)
def test_get_aws_credentials_file_from_env(environ, expected):
    with mock.patch.dict(os.environ, environ, clear=True):
        # ...
```

I also tend to steer away from the `monkeypatch` fixture due to not being able to tightly control when the fixture gets applied / unapplied and prefering explicit use of `mock.patch`, but I can let that go if it's too much of a nitpick :)